sklearn/preprocessing/tests/test_data.py

```diff
@@ -219,6 +219,12 @@ def test_standard_scaler_1d():
     assert_array_almost_equal(X_scaled.std(axis=0), .0)
     assert_equal(scaler.n_samples_seen_, X.shape[0])
 
+    # Test NaN Feature
+    X *= np.nan
```

Comment:
This is only a corner case. I would make 2 tests:

* one where all values are NaN.
* one where some values of a single column are NaN.